witness/src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -38,10 +41,25 @@ expandBlock ::
     => Bool
     -> Block
     -> ERoComp Exceptions Ids Values ctx SBlock
-expandBlock applyFees Block{..} = do
+expandBlock applyFees block@Block{..} = do
     let issuer = hIssuer bHeader
     stateTxs <- expandGTxs applyFees issuer (bbTxs bBody)
-    pure $ SD.Block bHeader (SPayload stateTxs $ hash bBody)
+    metaStateTx <- expandBlockMetaTx (BlockMetaTx block)
+    pure $ SD.Block bHeader (SPayload (metaStateTx : stateTxs) (hash bBody))
+
+expandBlockMetaTx
+    :: ( Default (ChgAccum ctx)
+       , HasLens ctx RestrictCtx
+       , HasLens ctx (ChgAccumCtx ctx)
+       , HasCoreConfig
+       )
+    => BlockMetaTx -> ERoComp Exceptions Ids Values ctx SStateTx
+expandBlockMetaTx meta =
+    let stateTx = StateTxType $ getId (Proxy @TxIds) BlockMetaTxTypeId
+        proof = BlockMetaTxWitnessProof
+    in expandUnionRawTxs (\_ -> (stateTx, proof, seqExpandersBlockMetaTx)) [meta] >>= \case
+           [expanded] -> return expanded
+           _          -> error "expandBlockMetaTx: did not expand 1 meta into 1 sd tx"
```

Comment:
Doesn't SD provide a function for expanding _one_ transaction? This looks ugly.

Reply:
There is definitely no one :snowdrop: